src/Collectors/TimelineDataCollector.php

```diff
@@ -2,25 +2,49 @@
 
 namespace AG\ElasticApmLaravel\Collectors;
 
-use AG\ElasticApmLaravel\Collectors\Interfaces\DataCollectorInterface;
+use AG\ElasticApmLaravel\Agent;
+use AG\ElasticApmLaravel\Contracts\DataCollector;
+use Illuminate\Foundation\Application;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\Log;
 
 /**
  * Collects info about the request duration as well as providing
  * a way to log duration of any operations.
  */
-class TimelineDataCollector implements DataCollectorInterface
+abstract class TimelineDataCollector implements DataCollector
 {
+    /** @var Application */
+    protected $app;
+
+    /** @var Agent */
+    protected $agent;
+
+    /** @var Collection */
     protected $started_measures;
+
+    /** @var Collection */
     protected $measures;
+
+    /** @var float */
     protected $request_start_time;
 
-    public function __construct(float $request_start_time)
+    public function __construct(Application $app, Agent $agent)
     {
+        $this->app = $app;
+        $this->agent = $agent;
+
         $this->started_measures = new Collection();
         $this->measures = new Collection();
-        $this->request_start_time = $request_start_time;
+
+        $this->request_start_time = $this->agent->getRequestStartTime();
+
+        $this->registerEventListeners();
+    }
+
+    public function getName(): string
```

Comment:
We might want to remove this implementation, to _force_ child classes to provide a name for the custom collector.